### PR TITLE
Add custom headers to all ElasticSearch requests

### PR DESCRIPTION
**Pull request by chrean**
## Description

Fixes [PLAT-932]

Adds following custom header to all requests sent to ElasticSearch

```
X-Client-Site-ID
X-Client-Env
```

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Check out PR.
1. Sandbox a site with VIP Search enabled (e.g. 1120)
1. How can we see whether those headers are really sent out or not? `var_dump()`ing shows they are actually used, but I don't know how we can be 100% sure they are making all their way to ES. 🤔

[PLAT-932]: https://vipjira.atlassian.net/browse/PLAT-932

### CODE REVIEW

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 1]
```diff
@@ -143,6 +143,8 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 
 		$timeout = $this->get_http_timeout_for_query( $query );
 
+		// Add custom headers to identify authorized traffic
+		$args['headers'] = array_merge( $args['headers'], array( 'X-Client-Site-ID' => get_current_blog_id(), 'X-Client-Env' => VIP_GO_ENV ) );
```
Comment (mjangda) [inline elasticsearch/includes/classes/class-elasticsearch.php]: Site ID should be the Go site ID not the WordPress site ID.

Comment (chrean) [reply elasticsearch/includes/classes/class-elasticsearch.php]: 7bb95bc2 if I got that right

Comment (mjangda) [reply elasticsearch/includes/classes/class-elasticsearch.php]: Oh, actually, I guess we should technically check that `$args['headers']` is an array first.

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 2]
```diff
@@ -143,6 +143,8 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 
 		$timeout = $this->get_http_timeout_for_query( $query );
 
+		// Add custom headers to identify authorized traffic
+		$args['headers'] = array_merge( $args['headers'], array( 'X-Client-Site-ID' => FILES_CLIENT_SITE_ID, 'X-Client-Env' => VIP_GO_ENV ) );
```
Comment (mjangda) [inline elasticsearch/includes/classes/class-elasticsearch.php]:
```suggestion
		if ( ! isset( $args['headers'] ) || ! is_array( $args['headers'] ) ) {
			$args['headers'] = [];
		}
		$args['headers'] = array_merge( $args['headers'], array( 'X-Client-Site-ID' => FILES_CLIENT_SITE_ID, 'X-Client-Env' => VIP_GO_ENV ) );
```

**Comment by mjangda** [comment]
r802-stacks

**Pull request merged**